### PR TITLE
Comment form

### PR DESCRIPTION
**Pull request by kevlartest**
Added an email, comment form on the home page that sends content as a POST request to the Servlet backend, where it is stored in a HashMap.

Screenshot 2020-07-20 at 3 51 53 PM

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -16,14 +16,28 @@
...
+        <div id="quote-container"></div>
+
+        <hr>
+        <form action="/comments" method="POST">
+            <h3>Add a Comment:</h3>
+            <label for="email">Email:</label>
+            <input type="email" id="email" name="email">
+            <br/><br/>
+            <label for="body">Comment:</label>
+            <input type="text" id ="body" name="body">
```
Comment (gjrb) [inline portfolio/src/main/webapp/index.html:35]: id="body"

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -16,14 +16,28 @@
...
+            <br/><br/>
+
+            <input type="submit" />
```
Comment (gjrb) [inline portfolio/src/main/webapp/index.html:38]: No need to close this: https://google.github.io/styleguide/htmlcssguide.html#Document_Type

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -16,14 +16,28 @@
...
+
+        <hr>
+        <form action="/comments" method="POST">
+            <h3>Add a Comment:</h3>
+            <label for="email">Email:</label>
+            <input type="email" id="email" name="email">
+            <br/><br/>
+            <label for="body">Comment:</label>
+            <input type="text" id ="body" name="body">
+            <br/><br/>
```
Comment (gjrb) [inline portfolio/src/main/webapp/index.html:36]: No need to close the <br> https://google.github.io/styleguide/htmlcssguide.html#Document_Type

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 4]
```diff
@@ -15,18 +15,35 @@
...
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        String email = request.getParameter("email");
+        String body = request.getParameter("body");
+
+        System.out.println("Email: " + email + " Comment: " + body);
+
+        comments.put(email,body);
```
Comment (gjrb) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:45]: Would this limit you to a single comment per email address? Everytime you added a new comment the previous one would be overwritten?

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 5]
```diff
@@ -15,18 +15,35 @@
...
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+  private Map<String,String> comments;
+
+    @Override
+    public void init() {
+        comments = new HashMap<>();
```
Comment (gjrb) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:34]: Should you define the type for the hashmap, I think you can do HashMap<String, String>

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java (hunk 4 above)
Comment (kevlartest) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:45]: Yes, you're right. The map should contain a list of comments, not a single comment.

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java (hunk 5 above)
Comment (kevlartest) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:34]: It is just being initialised there, as it was already defined above. It isn't required to specify the type if it was already specified, I don't seem to find anything in the style guide related to omitting the type

Comment (gjrb) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:36]: Ah I totally missed the above, that is fine

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+        final Comment comment = new Comment(email,body,timestamp);
+
+        System.out.println("Timestamp: " + timestamp + " Email: " + email + " Comment: " + body);
+
+        if(!comments.containsKey(email)){
+            comments.put(email, new ArrayList<>());
+        }
+
+        comments.get(email).add(comment);
```
Comment (gjrb) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:54]: One thing to note here is that all of your comments will be bundled by email address, how will this impact your data if you want to display them on the page sorted by timestamp or for instance? Or are you planning on displaying them grouped by email?

Comment (kevlartest) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:54]: I have changed this in the *datastore* branch, where the comments are permantently stored in a database and thus can be sorted by timestamp or filtered by email.

**Pull request merged**